File: vanilla-mvp/src/main/java/be/vergauwen/simon/cleanmvp/ui/TestPresenter.java
package be.vergauwen.simon.cleanmvp.ui;

import android.util.Log;
import be.vergauwen.simon.himura.MVPPresenter;

public class TestPresenter extends MVPPresenter<TestContract.View>
    implements TestContract.Presenter {

    @Override
    public void loadThings() {
        Log.e("TestPresenter",".loadThings()");
        getView().showThings();
    }
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/ui/TestPresenter.kt
package be.vergauwen.simon.kotlindaggermvp.ui

import android.content.Context
import android.util.Log
import be.vergauwen.simon.himurakotlin.MVPPresenter
import javax.inject.Inject


class TestPresenter
@Inject
constructor(private val context: Context) : MVPPresenter<TestContract.View>(), TestContract.Presenter<TestContract.View> {

  override fun loadThings() {
    Log.e("TestPresenter", ".loadThings()")
    getView()?.showThings()
  }
}